superset-frontend/src/components/DropdownContainer/index.tsx

```diff
@@ -0,0 +1,233 @@
+    if (container) {
+      const { children } = container;
+      const childrenArray = Array.from(children);
+
+      // Stores items width once
+      if (itemsWidth.length === 0) {
+        setItemsWidth(
+          childrenArray.map(child => child.getBoundingClientRect().width),
+        );
+      }
+
+      // Calculates the index of the first overflowed element
+      const index = childrenArray.findIndex(
+        child =>
+          child.getBoundingClientRect().right >
+          container.getBoundingClientRect().right,
+      );
+      setOverflowingIndex(index === -1 ? children.length : index);
+
+      if (width > previousWidth && overflowingIndex !== -1) {
+        // Calculates remaining space in the container
+        const button = current?.children.item(1);
+        const buttonRight = button?.getBoundingClientRect().right || 0;
+        const containerRight = current?.getBoundingClientRect().right || 0;
+        const remainingSpace = containerRight - buttonRight;
+        // Checks if the first element in the popover fits in the remaining space
+        const fitsInRemainingSpace = remainingSpace >= itemsWidth[0];
+        if (fitsInRemainingSpace && overflowingIndex < items.length) {
+          // Moves element from popover to container
+          setOverflowingIndex(overflowingIndex + 1);
+        }
+      }
+    }
+  }, [
+    current,
+    items.length,
+    itemsWidth,
+    overflowingIndex,
+    previousWidth,
+    width,
+  ]);
+
+  const notOverflowedItems = useMemo(
+    () =>
+      items
+        .slice(0, overflowingIndex !== -1 ? overflowingIndex : items.length)
+        .map(item => React.cloneElement(item.element, { key: item.id })),
+    [items, overflowingIndex],
+  );
+
+  const notOverflowedIds = useMemo(
+    () =>
+      items
+        .slice(0, overflowingIndex !== -1 ? overflowingIndex : items.length)
+        .map(item => item.id),
+    [items, overflowingIndex],
+  );
+
+  const overflowedItems = useMemo(
+    () =>
+      overflowingIndex !== -1
+        ? items
+            .slice(overflowingIndex, items.length)
+            .map(item => React.cloneElement(item.element, { key: item.id }))
+        : [],
+    [items, overflowingIndex],
+  );
+
+  const overflowedIds = useMemo(
+    () =>
+      overflowingIndex !== -1
+        ? items.slice(overflowingIndex, items.length).map(item => item.id)
+        : [],
+    [items, overflowingIndex],
+  );
+
+  useEffect(() => {
+    if (onOverflowingStateChange) {
+      onOverflowingStateChange({
+        notOverflowed: notOverflowedIds,
+        overflowed: overflowedIds,
+      });
+    }
+  }, [notOverflowedIds, onOverflowingStateChange, overflowedIds]);
+
+  const popoverContent = useMemo(
+    () => (
+      <div
+        css={css`
+          display: flex;
+          flex-direction: column;
+          gap: ${theme.gridUnit * 3}px;
+          width: 200px;
+        `}
+        style={popoverStyle}
+      >
+        {overflowedItems}
```

Comment:
Can we make the popover content customizable via props as well? In the case of horizontal filter bar we'll need to hide filters out of scope in a collapsible inside of popover

Reply:
I added the `popoverContent` property 👍🏼